### PR TITLE
Issue 1241: Migration - Registration

### PR DESCRIPTION
**Pull request by dleszcz**
**Description**
- Create a registration container
- Create register request/verify components
- Move styles to styled components
- Remove old unused files

**Bugs**

- [x] Can we display the message of password needs to be at least x characters If the password is rejected because of that? Password is weak should only be use when it's genuinely weak.
- [x] UK should be at the top for convenience, as most of our customer are UK based.
- [x] Can we make this list smaller (it's scrollable anyway)
- [ ] Typing on the keyboard should jump to that letter, see angular version:
- [x] Make sign up for latest news checked by default
- [x] We should only allow letters, numbers, underscores, hypens on username. This one should be flagged
- [x] I don't think we should do this message via snack bar. Can we make it so it's like before? (ie. the form changes to the message). Reason being the user may have missed it and will try to request again, or it will make it easy for a user to keep trying brute force to trigger something.
- [x] Similar case for this one. Especially their token will be invalid and pressing again will cause an error message to come up.
- [x] Clicking on logo should always redirect back to login page (unless session is active, in which case dashboard.)
- [x] Also, the sign-up page dies if I have recaptcha configured:

### CODE REVIEW

File: frontend/routes/registerVerify/registerVerify.component.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	};
+
+	public componentDidMount() {
+		const { username, token, pay} = queryString.parse(this.props.location.search);
+		const { verifyMessage } = this.props;
+
+		if (verifyMessage) {
+			this.setState({ verifyMessage });
+		}
+		if (!username || !token) {
```
Comment (simplypixi) [inline frontend/routes/registerVerify/registerVerify.component.tsx]: Could you set changes in one object to call setState one time?

File: frontend/routes/registerVerify/registerVerify.component.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+			<Container
+				container
+				direction="column"
+				justify="center"
+				alignItems="center">
+				<Logo />
+				<Grid item xs={9} sm={6} md={4} lg={3} xl={2}>
+					<Panel title="Registered for 3D Repo">
+						<Paragraph>
+							{verifyMessage && verifyMessage}
```
Comment (simplypixi) [inline frontend/routes/registerVerify/registerVerify.component.tsx]: Is it correct? :)

File: frontend/routes/signUp/components/reCaptcha/reCaptcha.component.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+				target: {
+					value: this.state.value,
+					name: this.props.name
+				}
+			});
+		}
+	}
+
+	public onReCaptchaChange = (value) => {
+		this.setState({ value}, this.handleChange);
```
Comment (simplypixi) [inline frontend/routes/signUp/components/reCaptcha/reCaptcha.component.tsx]: missing space

File: frontend/routes/signUp/components/reCaptcha/reCaptcha.component.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+	public render() {
+		const { onBlur, key } = this.props;
+
+		return (
+			<Container>
+				<ReCAPTCHA
+						onBlur={onBlur}
+						sitekey={key}
+						onChange={(value) => this.onReCaptchaChange(value)}
```
Comment (simplypixi) [inline frontend/routes/signUp/components/reCaptcha/reCaptcha.component.tsx]: Unnecessary arrow function. You can pass `onReCaptchaChange`directly.

File: frontend/routes/signUp/signUp.component.tsx [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+		return (
+			<Container
+				container
+				direction="column"
+				justify="center"
+				alignItems="center">
+				<Logo />
+				<StyledGrid item xs={9} sm={6} md={6} lg={6} xl={2}>
+					<Panel title="Sign up">
+						<Headline>{'Creating a 3D Repo account is free'}</Headline>
```
Comment (simplypixi) [inline frontend/routes/signUp/signUp.component.tsx]: You can remove these brackets

File: frontend/routes/signUp/signUp.styles.ts [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+  && {
+    margin-top: 16px;
+    margin-bottom: 8px;
+    margin-left: 12px;
+  }
+`;
+
+export const ButtonContainer = styled.div`
+  display: flex;
+	justify-content: flex-end;
```
Comment (simplypixi) [inline frontend/routes/signUp/signUp.styles.ts]: wrong indentation

Comment (simplypixi) [reply frontend/routes/signUp/signUp.styles.ts]: please, rebase this branch into #1183 and then check linter results

Comment (dleszcz) [reply frontend/routes/signUp/signUp.styles.ts]: I think this indentation is a github issue.
screenshot 2018-11-21 at 17 53 29


Rebase and PR fixes done.

**Comment by carmenfan** [comment]
@dleszcz @simplypixi  I changed the merge to #1183 as #1233 has merged into #1183.

Can you guys fix the conflicts?

File: frontend/routes/registerVerify/registerVerify.component.tsx [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+		if (missingParams || verifyMessage) {
+			this.setState({
+				verifyMessage: missingParams ? missingParamsMessage : verifyMessage
+			});
+		} else {
+			this.props.verifyRequest(username, token);
+		}
+	}
+	public componentDidUpdate(prevProps) {
```
Comment (sanmont3drepo) [inline frontend/routes/registerVerify/registerVerify.component.tsx]: A new line before the function here?

File: frontend/routes/registerVerify/registerVerify.component.tsx [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+		const { verifyMessage } = this.props;
+
+		if (prevProps.verifyMessage !== verifyMessage) {
+			this.setState({ verifyMessage });
+		}
+	}
+	public render() {
```
Comment (sanmont3drepo) [inline frontend/routes/registerVerify/registerVerify.component.tsx]: new line?

File: frontend/routes/signUp/signUp.component.tsx [hunk 9]
```diff
@@ -0,0 +1,320 @@
...
+
+export class SignUp extends React.PureComponent<IProps, IState> {
+	public state = {
+		passwordStrengthMessage: ''
+	};
+
+	public reCaptchaWrapperRef = React.createRef<any>();
+
+	public handleSubmit = (values, form) => {
+		const data = {
```
Comment (sanmont3drepo) [inline frontend/routes/signUp/signUp.component.tsx]: Could this be changed to:
```suggestion
		const data = {...values};
```

Or there are more data that's being passed that you dont want?

Comment (dleszcz) [reply frontend/routes/signUp/signUp.component.tsx]: Good point! Exactly, I used lodash omit function for this one.

**Comment by dleszcz** [comment]
@sanmont3drepo Have you finished checking this PR? 
If yes, I think @carmenfan can look at it

**Comment by carmenfan** [comment]
@dleszcz  can you rebase/merge with staging? :)

**Comment by simplypixi** [comment]
@carmenfan Rebased ;)

**Comment by carmenfan** [comment]
Also, the sign-up page dies if I have recaptcha configured:
image

**Comment by simplypixi** [comment]
@carmenfan Ready to re-check ;)

**Comment by simplypixi** [comment]
@carmenfan Should be fixed ;)

**Pull request merged**